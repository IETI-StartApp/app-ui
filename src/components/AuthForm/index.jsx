import React, {useRef, useState} from 'react';
import {useHistory} from "react-router";
import {Button, Checkbox, TextField, Typography} from "@material-ui/core";
import 'typeface-roboto';
import CircleChecked from '@material-ui/icons/CheckCircleOutline';
import CircleUnchecked from '@material-ui/icons/RadioButtonUnchecked';

import {CheckBoxArea, FooterWrapper, ForgotPasswdArea, InputArea, TextArea, Wrapper} from "./styles";
import google from '../../icons/google.svg'
import {Auth} from "../../authServices/firebase-config";
import {CustomTypography} from "../index";
import { getUserByEmail } from '../../services/userServices';

export const AuthForm = ({functions}) => {
    const {signInWithEmailAndPassword, signInWithGoogle} = functions
    const [checked, setChecked] = useState(false);
    const emailRef = useRef('');
    const passwdRef = useRef('');
    const history = useHistory();

    function handleChange(event) {
        setChecked(event.target.checked)
    }
    const redirectGoogle = async () =>{
        await signInWithGoogle();
        const user = await getUserByEmail(Auth.currentUser.email);
        console.log(user)
        redirect(user,Auth.currentUser)
    } 
    const handleSubmit = async (e) => {
        e.preventDefault()
        try {
            await signInWithEmailAndPassword(emailRef.current.value, passwdRef.current.value)
            const user = await getUserByEmail(Auth.currentUser.email);
            redirect(user,Auth.currentUser)
        } catch (error) {
            alert(error)
        }
    }
    const redirect = (user, auth) =>{
        if (user && auth) {
            history.replace('/dashboard',{ role: user.role})
        }else{
            history.replace("/complete-info")
        }
    }
    return (
        <Wrapper>
            <form onSubmit={handleSubmit}>
                <TextArea>
                    <Typography>
                        Email
                    </Typography>
                </TextArea>
                <InputArea>
                    <TextField inputRef={emailRef} label="Email" variant="outlined" fullWidth type='email' required/>
                </InputArea>
                <TextArea>
                    <Typography>
                        Contrase単a
                    </Typography>
                </TextArea>
                <InputArea>
                    <TextField inputRef={passwdRef} label='Contrase単a' variant="outlined" fullWidth type='password'
                               required/>
                </InputArea>
                <FooterWrapper>
                    <CheckBoxArea>
                        <Checkbox
                            className={`without-padding`}
                            checked={checked}
                            onChange={handleChange}
                            style={{
                                color: "#4A5568",
                                background: "none",
                                borderradius: "none"
                            }}
                            icon={<CircleUnchecked/>}
                            checkedIcon={<CircleChecked/>}
                        />
                        <CustomTypography text='#2D3748' style={{padding: '10px 0'}}>Recordarme</CustomTypography>
                    </CheckBoxArea>
                    <ForgotPasswdArea>
                        <CustomTypography
                            style={{padding: '10px 0', cursor: 'pointer'}}
                            text='#2C5282'
                        >
                            多Olvidaste la contrase単a?
                        </CustomTypography>
                    </ForgotPasswdArea>
                </FooterWrapper>
                <Button
                    type='submit'
                    fullWidth
                    style={{
                        background: "#950740",
                        color: 'white',
                        textTransform: 'none',
                        height: '50px'
                    }}>
                    Entrar
                </Button>
            </form>
            <Button onClick={redirectGoogle}
                    fullWidth
                    style={{
                        background: "#2D3748",
                        color: 'white',
                        textTransform: 'none',
                        height: '50px',
                        marginTop: '20px'
                    }}>
                <img src={google} height={20} width={20} style={{paddingRight: '11px'}} alt='logo'/>
                Entrar con Google
            </Button>
        </Wrapper>
    );

};
