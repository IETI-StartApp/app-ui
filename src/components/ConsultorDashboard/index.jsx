import React, {useState,useEffect} from 'react';
import Navigation from '../Navbar/Navigation';
import { TicketCard } from '../Support/TicketCard';
import { useStyles } from './styles';
import { Grid } from '@material-ui/core/';
import {getAllTickets} from '../../services/ticketServices'
import {Auth} from "../../authServices/firebase-config";
export const ConsultorDashboard = ({testing, objects}) => {
    const classes = useStyles();
    const [allTickets, setTickets] = useState([]);
    useEffect(() => {
        async function fetchTickets(){
            const receivedTickets = testing?objects : await getAllTickets();
            setTickets(receivedTickets);
        }
        fetchTickets();
    }, [allTickets,testing,objects]);

    return (
        <>
            <Navigation />
            <div className={classes.paper}>
            <Grid
                container
                direction="row"
                justify="flex-start"
                alignItems="center"
                spacing={1}
            >
                <Grid item sm={8} md={6}>
                    <TicketCard tickets={allTickets} title={"Tickets disponibles"} status = "OPEN"/>
                </Grid>
                
                <Grid item sm={8} md={6}>
                    <TicketCard tickets={allTickets.filter(ticket => ticket.receptor?.email === Auth.currentUser.email)} title={"Mis tickets pendientes"} status = "PENDING"/>
                </Grid>
            </Grid>
                
            </div>
            
        </>
    );
}
