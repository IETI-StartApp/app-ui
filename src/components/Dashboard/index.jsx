import React from 'react';
import Navigation from '../Navbar/Navigation';
import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline';
import { Grid, Fab } from '@material-ui/core/';
import { useStyles } from './styles';
import { ItemsGrid } from './ItemsGrid';
import { ProjectsCard } from './ProjectsCard';
import { ChartCard } from './ChartCard';

export const Dashboard = ( { items, projects, chart, projectsTitle, chartTitle } ) => {
    const classes = useStyles();

    return (
        <>
            <Navigation />
            <div className={classes.offset}></div>
            <div className={classes.root}>
                <Grid container spacing={2} direction="col" justify="center" alignItems="center" className={classes.allItems}>
                    <Grid item xs={12}>
                        <ItemsGrid items={items} />
                    </Grid>
                    <Grid item xs={12}>
                        <Grid container spacing={2} direction="row" justify="center">
                            <Grid item xs={12} sm={10} lg={6}>
                                <ChartCard title= { chartTitle } chart = { chart } />
                            </Grid>
                            <Grid item xs={12} sm={10} lg={6}>
                                <ProjectsCard title={ projectsTitle } projects={ projects } />
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
                <Fab aria-label="Chat" className={classes.fab}>
                    <ChatBubbleOutlineIcon />
                </Fab>
            </div>
        </>
    );
}