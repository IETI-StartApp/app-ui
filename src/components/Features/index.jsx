import React from "react";

export const Features = (props) => {
  return (
    <div id='features' className='text-center'>
      <div className='container'>
        <div className='col-md-10 col-md-offset-1 section-title'>
          <h2>Recursos para ayudarte a recaudar fondos</h2>
        </div>
        <div className='row'>
          {props.data
            ? props.data.map((d, i) => (
                <div key={`${d.title}-${i}`} className='col-xs-4'>
                  {' '}
                  <i className={d.icon}/>
                  <h3>{d.title}</h3>
                  <p>{d.text}</p>
                </div>
              ))
            : 'Loading...'}
        </div>
      </div>
    </div>
  )
}
