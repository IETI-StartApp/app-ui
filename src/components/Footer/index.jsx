import React from 'react';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';

const Copyright = () => {
    return (
        <Typography variant="body2" color="textSecondary" align="center">
            {'Copyright Â© '}
            <Link color="inherit" href="https://material-ui.com/">
                StartApp
            </Link>{' '}
            {new Date().getFullYear()}
            {'.'}
        </Typography>
    );
}

export const Footer = ({ title}) => {

    return (
        <footer style={{
            backgroundColor: 'transparent',
            padding: '6px',
        }}>
            <Container maxWidth="lg">
                <Typography variant="h6" align="center" gutterBottom>
                    {title}
                </Typography>
                <Copyright/>
            </Container>
        </footer>
    );
}
