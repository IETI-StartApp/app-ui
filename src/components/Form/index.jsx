import React, {useRef, useState} from 'react';
import {Button, Checkbox, TextField, Typography} from "@material-ui/core";
import 'typeface-roboto';
import CircleChecked from '@material-ui/icons/CheckCircleOutline';
import CircleUnchecked from '@material-ui/icons/RadioButtonUnchecked';

import {CheckBoxArea, FooterWrapper, ForgotPasswdArea, InputArea, TextArea, Wrapper} from "./styles";
import {CustomTypography} from "../ButtonCustom";
import google from '../../icons/google.svg'
import {useAuth} from "../../services/Auth";
import {useHistory} from "react-router";


export const Form = ({signInWithEmailAndPassword, signInWithGoogle}) => {

    const [checked, setChecked] = useState(false);
    const emailRef = useRef('');
    const passwdRef = useRef('');
    const history = useHistory();

    function handleChange(event) {
        setChecked(event.target.checked)
    }

    const handleSubmit = async (e) => {
        e.preventDefault()
        try {
            console.log(emailRef.current.value, passwdRef.current.value)
            await signInWithEmailAndPassword(emailRef.current.value, passwdRef.current.value)
            history.push('/register-project')
        } catch (error) {
            alert(error)
        }
    }
    return (
        <Wrapper>
            <form onSubmit={handleSubmit}>
                <TextArea>
                    <Typography>
                        Email
                    </Typography>
                </TextArea>
                <InputArea>
                    <TextField inputRef={emailRef} label="Email" variant="outlined" fullWidth type='email' required/>
                </InputArea>
                <TextArea>
                    <Typography>
                        Contrase単a
                    </Typography>
                </TextArea>
                <InputArea>
                    <TextField inputRef={passwdRef} label='Contrase単a' variant="outlined" fullWidth type='password'
                               required/>
                </InputArea>
                <FooterWrapper>
                    <CheckBoxArea>
                        <Checkbox
                            className={`without-padding`}
                            checked={checked}
                            onChange={handleChange}
                            style={{
                                color: "#4A5568",
                                background: "none",
                                borderradius: "none"
                            }}
                            icon={<CircleUnchecked/>}
                            checkedIcon={<CircleChecked/>}
                        />
                        <CustomTypography text='#2D3748' style={{padding: '10px 0'}}>Recordarme</CustomTypography>
                    </CheckBoxArea>
                    <ForgotPasswdArea>
                        <CustomTypography
                            style={{padding: '10px 0', cursor: 'pointer'}}
                            text='#2C5282'
                            onClick={() => {
                                alert('You are going to be redirected!')
                            }}
                        >
                            多Olvidaste la contrase単a?
                        </CustomTypography>
                    </ForgotPasswdArea>
                </FooterWrapper>
                <Button
                    type='submit'
                    fullWidth
                    style={{
                        background: "#950740",
                        color: 'white',
                        textTransform: 'none',
                        height: '50px'
                    }}>
                    Entrar
                </Button>
            </form>
            <Button onClick={() => {
                signInWithGoogle()
                history.push('/register-project')
            }}
                    fullWidth
                    style={{
                        background: "#2D3748",
                        color: 'white',
                        textTransform: 'none',
                        height: '50px',
                        marginTop: '20px'
                    }}>
                <img src={google} height={20} width={20} style={{paddingRight: '11px'}} alt='logo'/>
                Entrar con Google
            </Button>
        </Wrapper>
    );

};
