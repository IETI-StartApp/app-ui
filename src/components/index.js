export * from './ButtonCustom'
export * from './AuthForm';
export * from './ProjectItems'
export * from './BarInfo'
export * from './InformacionFinazas'
export * from './InformacioProyecto'
export * from './Navbar'
export * from '../authServices/PrivateRoute'
export * from './Progressbar'
export * from './Register'
export * from './Footer'
export * from './Header'
export * from './ItemContainer'
