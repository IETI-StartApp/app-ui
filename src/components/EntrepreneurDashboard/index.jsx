import React from 'react';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import { Dashboard } from '../Dashboard/index';
import { ProjectsChart } from './ProjectsChart';
import PermIdentityIcon from '@material-ui/icons/PermIdentity';
import CheckIcon from '@material-ui/icons/Check';

export const EntrepreneurDashboard = () => {
    const items = [
        { number: "5", description: "Proyectos Creados", icon: <StarBorderIcon fontSize="large" />, path: "/" },
        { number: "4.5M", description: "Capital recaudado", icon: <AttachMoneyIcon fontSize="large" />, path: "/dashboard" },
        { number: "17", description: "Socios", icon: <PermIdentityIcon fontSize="large" />, path: "/" },
        { number: "0", description: "Proyectos exitosos", icon: <CheckIcon fontSize="large" />, path: "/" }
    ];
    const projects = [
        {
            name: "Proyecto 1", 
            image:"https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg", 
            finance: { value:15000000, valuation:20000000, startDate: "10/03/2021"}
        },
        {
            name: "Proyecto 2", 
            image:"https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png", 
            finance: { value:10000000, valuation:17500000, startDate: "12/02/2021"},
        },
        {
            name: "Proyecto 3", 
            image:"https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg", 
            finance: { value:8000000, valuation:12000000, startDate: "21/01/2021"},
        },
    ];

    return (
        <Dashboard items={ items } projects={ projects } chart={ <ProjectsChart /> } projectsTitle={ "Mis proyectos" } chartTitle={ "Mis proyectos: porcentaje financiado" } />
    );
}