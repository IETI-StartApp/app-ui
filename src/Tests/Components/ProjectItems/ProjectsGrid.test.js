import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import React from 'react';
import {ProjectsItems} from "../../../components";

describe('Testing projects grid', () => {

    const wrapper = shallow(<ProjectsItems
        projects={[
            {
                name: "a",
                image: "a",
                finance: {}
            },
            {
                name: "a",
                image: "a",
                finance: {}
            },
            {
                name: "a",
                image: "a",
                finance: {}
            }]}/>)

    test('should load CardProject', () => {
        expect(wrapper).toMatchSnapshot();
    })

})
