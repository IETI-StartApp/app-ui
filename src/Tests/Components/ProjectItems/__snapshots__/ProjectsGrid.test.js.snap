// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing projects grid should load CardProject 1`] = `
<Fragment>
  <Navigation />
  <WithStyles(ForwardRef(Container))
    className="makeStyles-cardGrid-4"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      key="project-items-key"
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject
          project={
            Object {
              "finance": Object {},
              "image": "a",
              "name": "a",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject
          project={
            Object {
              "finance": Object {},
              "image": "a",
              "name": "a",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={4}
        sm={6}
        xs={12}
      >
        <CardProject
          project={
            Object {
              "finance": Object {},
              "image": "a",
              "name": "a",
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
